gem/ci/build.ps1

```diff
@@ -0,0 +1,217 @@
+$ErrorActionPreference = 'Stop'
+
+function Get-CurrentDirectory
+{
+    $thisName = $MyInvocation.MyCommand.Name
+    [IO.Path]::GetDirectoryName((Get-Content function:$thisName).File)
+}
+
+function Get-ContainerVersion
+{
+    # Don't need to fetch anything else when running in Azure CI, as that will
+    # already have pulled everything down, and more importantly git-fetch will
+    # block waiting for authorization. TF_BUILD is set by Azure CI.
+    if ($ENV:TF_BUILD -ine 'true')
+    {
+        # shallow repositories need to pull remaining code to `git describe` correctly
+        if (Test-Path "$(git rev-parse --git-dir)/shallow")
+        {
+            git pull --unshallow
+        }
+
+        # tags required for versioning
+        git fetch origin 'refs/tags/*:refs/tags/*'
+    }
+
+    (git describe) -replace '-.*', ''
+}
+
+function Lint-Dockerfile($Path)
+{
+    hadolint --ignore DL3008 --ignore DL3018 --ignore DL4000 --ignore DL4001 $Path
+}
+
+function Build-Container2(
```

Comment:
I think that we can axe this method and collapse into the other `Build-Container` method.

PowerShell has first class support for parameter sets - https://docs.microsoft.com/en-us/powershell/developer/cmdlet/cmdlet-parameter-sets

Reply:
Yep, definitely; this is still a WIP.

I'm pushing changes to this and then consuming them in other projects.

🚧 🏗 🚧